integrations/jira/jira/client.py

```diff
@@ -103,107 +177,97 @@ async def create_events_webhook(self, app_host: str) -> None:
             "events": WEBHOOK_EVENTS,
         }
 
-        webhook_create_response = await self.client.post(
-            f"{self.webhooks_url}", json=body
-        )
-        webhook_create_response.raise_for_status()
+        await self._send_api_request("POST", self.webhooks_url, json=body)
         logger.info("Ocean real time reporting webhook created")
 
     async def get_single_project(self, project_key: str) -> dict[str, Any]:
-        project_response = await self.client.get(
-            f"{self.api_url}/project/{project_key}"
+        return await self._send_api_request(
+            "GET", f"{self.api_url}/project/{project_key}"
         )
-        project_response.raise_for_status()
-        return project_response.json()
 
     async def get_paginated_projects(
-        self, params: dict[str, Any] = {}
+        self, params: dict[str, Any] | None = None
     ) -> AsyncGenerator[list[dict[str, Any]], None]:
         logger.info("Getting projects from Jira")
-
-        params.update(self._generate_base_req_params())
-
-        total_projects = (await self._get_paginated_projects(params))["total"]
-
-        if total_projects == 0:
-            logger.warning(
-                "Project query returned 0 projects, did you provide the correct Jira API credentials?"
-            )
-
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] <= total_projects:
-            logger.info(f"Current query position: {params['startAt']}/{total_projects}")
-            project_response_list = (await self._get_paginated_projects(params))[
-                "values"
-            ]
-            yield project_response_list
-            params["startAt"] += PAGE_SIZE
+        async for projects in self._get_paginated_data(
+            f"{self.api_url}/project/search", "values", initial_params=params
+        ):
+            yield projects
 
     async def get_single_issue(self, issue_key: str) -> dict[str, Any]:
-        issue_response = await self.client.get(f"{self.api_url}/issue/{issue_key}")
-        issue_response.raise_for_status()
-        return issue_response.json()
+        return await self._send_api_request("GET", f"{self.api_url}/issue/{issue_key}")
 
-    async def get_paginated_issues(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def get_paginated_issues(
+        self, jql: str | None = None
```

Comment:
Let this method take in the params itself rather than have to compose jql and all the stuff internally. This is to ensure significant change in the future when other attributes will be added to the issue kind selector.